spatialmath/geo_obstacle.go

```diff
@@ -143,14 +143,14 @@ func GeoPointToPose(point, origin *geo.Point) Pose {
 	}
 }
 
-func PoseToGeoPoint(relativeTo *geo.Point, p Pose) *geo.Point {
+func PoseToGeoPoint(relativeTo *geo.Point, p Pose, lastHeading float64) *geo.Point {
```

Comment:
Firstly; the addition of `lastHeading` here was simply to test that this correctly fixes the orientation issues; this can be implemented differently/better.

On to the math.

What this function is doing, is saying "starting at this lat/lon position, move this many meters towards the X axis, and that many meters towards the Y axis, defined right-handedly such that Y is the direction the base is pointing".

So whether adding 1 meter to the Y component of the pose increments or decrements, latitude or longitude or some combination thereof, depends on which way the base is pointing while at `relativeTo`. This is `lastHeading`.

> I thought pose would have the orientation information. Why do we also need the lastHeading?

The orientation in the Pose is the desired orientation after translating as specified by the pose; it will have no effect whatsoever on the final returned `geo.Point`. It is still useful, though, as it ought be used by the caller to create the `lastHeading` used by a subsequent call to `PoseToGeoPoint` that passes the return value of a previous call.

This becomes easier to keep track of if we swap to using `spatialmath.GeoPose`. See the first sentence of this comment.